src/Parameter.h

```diff
@@ -4,26 +4,33 @@
 /** \file
  * Defines the internal representation of parameters to halide piplines
  */
+#include <string>
 
-#include "Buffer.h"
-#include "Expr.h"
+#include "IntrusivePtr.h"
+#include "Type.h"
+#include "Util.h"                   // for HALIDE_NO_USER_CODE_INLINE
+#include "runtime/HalideRuntime.h"  // for HALIDE_ALWAYS_INLINE
 
 namespace Halide {
 
 struct ArgumentEstimates;
-class OutputImageParam;
+template<typename T>
+class Buffer;
+struct Expr;
+struct Type;
 
 namespace Internal {
 
 struct ParameterContents;
 
 /** A reference-counted handle to a parameter to a halide
  * pipeline. May be a scalar parameter or a buffer */
-class Parameter {
+class Parameter final {
```

Comment:
Why final?

Reply:
Whoops! Scalpel left in patient. (There was a dangling virtual dtor left over from long ago -- I wanted to be sure no one was still overriding it. Will remove.)